api/src/main/java/com/redhat/ipaas/api/IPaasEntity.java

```diff
@@ -15,9 +15,15 @@
  */
 package com.redhat.ipaas.api;
 
-public interface IPaasEntity {
+public abstract class IPaasEntity {
 
-	public String getId();
-	public void setId(String id);
+	abstract public String getId();
+	abstract public void setId(String id);
+	
+	@Override
+	public boolean equals(Object obj) {
```

Comment:
Seeing as we're using java 8, I'd prefer to use something like `list.stream().anyMatch(new IdPredicate())` (a utility function providing that).